pkg/crc/telemetry/telemetry.go

```diff
@@ -0,0 +1,29 @@
+package telemetry
+
+import (
+	"context"
+)
+
+type contextKey struct{}
+
+var key = contextKey{}
+
+type Properties struct {
+	Map map[string]interface{}
+}
+
+func (p *Properties) Set(name string, value interface{}) {
+	p.Map[name] = value
+}
+
+func ContextWithEmptyProperties(ctx context.Context) context.Context {
+	return context.WithValue(ctx, key, &Properties{Map: make(map[string]interface{})})
+}
+
+func PropertiesFromContext(ctx context.Context) *Properties {
+	value := ctx.Value(key)
```

Comment:
Quoting https://blog.golang.org/context
> Value allows a Context to carry request-scoped data. That data must be safe for simultaneous use by multiple goroutines.

I don't think `Properties` as it is now is going to be safe to be shared across multiple go routines. Overall, I'm not so convinced by the use of the `context` package here:

> Incoming requests to a server should create a Context, and outgoing calls to servers should accept a Context. 

We have no incoming/outgoing requests/calls here but we still use a `context`, not clear to me we should be using this here.

Reply:
I will make `Properties` thread-safe.

We don't have many choices. Either, it is a global map in a var at package level, or it is a context.
The benefit of context is that it can be used in the daemon API quite nicely: just pass a context extended with telemetry and at the end of the request, send to segment.

Reply:
The segment package already has a `github.com/code-ready/crc/pkg/crc/config.Config` instance, really not clear to me why we need to add context + telemetry dependencies to the config package when the telemetry code already has access to it.

Reply:
All config keys are not relevant to all events. cpus/memory are used to crc start, not crc status. 
Also, having config in segment package is not enough: in `crc config set <key>`, it's impossible to pass key to segment without either a global map or context.